### ISSUE TITLE
Blocking crash reporters, non-ad related analytics by default?

### ISSUE THREAD

**Issue by bhouston** [uBlock-LLC/uBlock#564]
I run a site with no ads that is free: https://clara.io   With uBlock blocking google-analytics, (as well as piwik, pingdom, customer.io, mixpanel, kissmetrics, pingdom) by default, I will see a decrease in measured traffic.  I do not see why it is necessary to block the standard website analytics tools by default.  Webmasters such as myself use this to do quality control, improve the site by measuring usage, tracking errors, etc.

Blocking Google Analytics without stating this to users who just want to block ads is hurtful to all websites who want visitors and users.

Why not make it optional (opt in) to excluding standard statistical website analytic tools like Google Analytics?  Google analytics (and these other analytics) is not an Ad server by itself, Adwords is a separate product.

Also uBlock is blocking crash reporters like Bugsnag and NewRelic.  It is also blocking self-hosted analytics tools like PiWik (which doesn't track you across different sites or report into a Google-like mothership.)

This excessive blocking is outright harmful to web developers.

**Comment by gorhill** [comment]
> Blocking Google Analytics without stating this to users

It is disclosed up front that EasyPrivacy/Peter Lowe's are enabled out of the box.

> Google analytics is not an Ad server by itself

uBlock is not a "ad blocker". It's a tool for users to do whatever they want, including not using EasyPrivacy if they wish.

**Comment by bhouston** [comment]
> uBlock is not a "ad blocker".

The general public, and myself included until today, views uBlock as a better Ad-blocker because it doesn't use excessive memory, for example:
- http://lifehacker.com/ublock-is-a-fast-and-lightweight-alternative-to-adblock-1625246461
- http://www.reddit.com/r/firefox/comments/2siygp/first_version_of_popular_chrome_lightweight/

The primary people installing uBlock will be people looking to block ads because that concern is a lot more widespread than privacy tracking because ads are obvious and intrusive.  Privacy advocates are an important demographic but they pale in comparison to those that want an adblocker better than ABP.

> It is disclosed up front that EasyPrivacy/Peter Lowe's are enabled out of the box.

That isn't disclosure by any useful measure -- it is disclosure like a click-thru EULA.  No one except a very few know what "EasyPrivacy/Peter Lowe's" means.  This is a cop-out.

> It's a tool for users to do whatever they want, including not using EasyPrivacy if they wish.

You guys also block bugsnag (https://bugsnag.com/), and piwik (http://piwik.org/).  So I can not even switch from the centralized Google-based Analytics to a private piwik install -- thus you are not against Google but against analytics in general.

I think you guys do not understand how hard it is to make reliable JavaScript-based applications.  If you block analytics and bug reporting, you are hurting our ability to deliver quality.

It is shortsighted projects like this one that hurt the web.  If I was developing an Andriod application I could use analytics and crash reporting to create a great app.  But because I chose to do it more openly on the web, I've got to deal with Ad-blocker overreaching into blocking critical tools that I need.

For those that just say track HTTP accesses on the server, I can't!  I am running a complex JavaScript single page application were people have 40+ minute sessions on a single page.  Blocking crash reporting and JavaScript-based analytics is totally screws us over.

Of course you do not care, because crash reporting and even private analytics is apparently a privacy invasion.  I think you are blocking things (bugsnag, piwik) just because you can, not because it is the right thing to do.  To block analytics and bugsnag is to fundamentally misunderstand modern advanced web development -- and to say it doesn't matter or against your ideology is basically giving us the finger, essentially you are saying: screws us guys trying to push the boundaries of what can be done on the web.

I'm reminded of that quote: 

> “Ideology knows the answer before the question has been asked. Principles are something different: a set of values that have to be adapted to circumstances but not compromised away.” 

A principled approach would be to look at what your general users want -- they want an efficient adblocker.  A subset of your users want enhanced privacy.  You should meet both of those users needs through appropriate defaults and opt-in rather than forcing a rigid unthinking ideology on everyone trying to get away from ABP so that minimize the havok you rain down on unsuspecting victim such as ourselves.

**Comment by bhouston** [comment]
I can believe these are blocked by default -- this is declaring war on all website analytics tools used primarily for quality and traffic purposes, and not used for tracking customers across the web:
- bugsnag - crash reporting
- piwik - analytics
- mixpanel - analytics
- google analytics - analytics
- Kiss metrics - analytics
- customer.io - analytics
- optimizely.com - analytics
- newrelic.com - crash reporting
- seomoz.org - SEO
- pingdom.net - page load

It sucks that it is now considered to be appropriate to block these tools.  This hurts websites and puts them in opposition to uBlock, when really we shouldn't be in opposition.  I do not run ads, I do not use pixel trackers, or retargeting or anything like that.  I just use analytics and crash reporting.

The only one above that one can really be against is GA, but the rest do not have that reach and no run by companies that have ad networks.

Blocking analytics and crash reporting should not be the default of a tool that is rapidly gaining popularity for blocking ads -- it is a Trojan horse that hurts independent developers.  Now I have to write by own custom systems to replicate this functionality because if it isn't custom uBlock blocks it.

**Comment by gorhill** [comment]
You were given access to tools to track users, you used them. Users are given tools to prevent being tracked<sup>[1]</sup>, they use them. No wall of text is going to change this.

<sub>[1] uBlock, Ghostery, Disconnect, Privacy Badger, Adblock Plus, AdBlock, Adguard, Tor, VPN, proxies, blocking 3rd-party cookies, etc.</sub>

**Comment by bhouston** [comment]
> You were given access to tools to track users, you used them.

You enable it by default in what is rapidly replacing the main adblocker without disclosure.  This is the new normal.

Your blocking of crash reporters and self-analytics like piwik by default is wrong and straight out harmful to other web developers -- it means that I can not even change from GA to Piwik -- that is already blocked.  Thus the new emerging normal on the web is harmful to other web developers and you are saying you don't care.

> Users are given tools to prevent being tracked[1], they use them.

ABP/AB, Tor, AdBlock, VPN, Proxies, and third party cookie blockers do not block these sites (crash reporters and Piwik) by default.  The other tools are explicit that they block everything.  I'm a long time ABP/AB user as well.  I noticed the blocking of these things because it blocked it during dev after I switched to uBlock.  uBlock is misleading and it outright harmful to other web developers because of this.

> No wall of text is going to change this.

That is your response to real concerns of hurting other web developers?  tl/dr?  Seriously?

**Comment by gorhill** [comment]
> hurting other web developers

You are only offering opinions, none of them supported by any data, let alone rigorous data. Your opinions do not magically transform into facts because you feel strong about them. And even if you were to substantiate the highly dubious claim that not opting in to Google Analytics tracking "hurts web developers", it would still come down to users' interests vs. web developers' interests. uBlock is meant to serve users' interests, that's how I want it to be.

**Comment by bhouston** [comment]
Blocking crash reporters and even piwik hurts developers.  It isn't an opinion.  You are convinced I do not matter so feel free to continue to prejudge me and degrade what I am saying into an "opinion."  What I am I dealing with here - an 8 year old with their fingers covering their ears?

I know because we rely upon that crash reporting and analytics extensively to debug and prioritize our massive JavaScript application -- which does full 3D modeling, animation and rendering in the web browser.  It is a single page app and it has insane JavaScript.  uBlock makes me blind to what is going on, making debugging impossible.  I can not even switch away from GA to a private self-hosted analytics tool like Piwik because that is blocked too by default by uBlock.  Thus you are shutting off all avenues to create a quality application.  That is directly hurting me -- and I don't even run ads.  It isn't an opinion at all.

**Comment by gorhill** [comment]
It's not clear what you are trying to accomplish here. You made your point, I made mine, issue is closed. To keep repeating over and over your feelings about this accomplishes nothing.

**Comment by bhouston** [comment]
> You made your point, I made mine, issue is closed. 

You haven't acknowledged that you are hurting my website at all - you are in denial and avoiding dealing with the hard questions face on:

Why block piwik and crash reporters?  I don't have ads and I am not part of a tracker network.  Can you give a reason for blocking crash reporters (BugSnag, NewRelic) and locally hosted analytics like Piwik?

Blocking those means that even if I switched away from a third party hosted GA, I can not escape the damage that uBlock is doing to my ability to deliver a quality website.

tl/dr: uBlock considered harmful to web developers because of its excessive blocking by default.

**Comment by gorhill** [comment]
> Why block piwik and crash reporters?

Go ask EasyPrivacy maintainers: https://forums.lanik.us/viewforum.php?f=64

**Comment by gorhill** [comment]
Also of interest in the context: ["4 Simple Changes to Stop Online Tracking"](https://www.eff.org/deeplinks/2012/04/4-simple-changes-protect-your-privacy-online) (eff.org)

> Get Adblock Plus. After it is installed, be sure to change your filter preferences to add EasyPrivacy.

**Comment by Cephel** [comment]
> this is declaring war on all website analytics tools

Hopefully, yes.
